src/rgw/rgw_sal_motr.cc

```diff
@@ -1034,11 +1035,37 @@ int MotrBucket::list(const DoutPrefixProvider *dpp, ListParams& params, int max,
       rgw_bucket_dir_entry ent;
       auto iter = vals[i].cbegin();
       ent.decode(iter);
+      std::string key_name = ent.key.name + "/null";
```

Comment:
initialize key_name before for loop

Reply:
The `ent `structure gets decoded value in the for loop, hence this initialization is required here.

Reply:
ok got it